indra/llcommon/llstring.h

```diff
@@ -1596,6 +1597,15 @@ void LLStringUtilBase<T>::replaceTabsWithSpaces( string_type& str, size_type spa
 	str = out_str;
 }
 
+//static
+template<class T>
+std::basic_string<T> LLStringUtilBase<T>::capitalize(const string_type& str)
```

Comment:
we already have basic 'case' operatiions, why not use LLStringUtil::toUpper?

Reply:
toUpper makes "abc def" => "ABC DEF"
capitalize makes "abc def" => "Abc Def"
we have a procedure capitalize modifying it's argument in-place
but sometimes it is more usable to have a function returning the result but not modifying the argument
I added a function capitalize which is just a wrapper around the procedure capitalize (a syntax sugar)

Reply:
Please add a comment before the function to explain what it does.

Reply:
As you can see, I didn't add any new functionality, the function capitalize() was introduced earlier with no comments
I just overloaded this function for the argument of type a constant reference to the string
![image](https://github.com/secondlife/viewer/assets/124201357/df4021ff-690b-48c4-922f-78fdb2be381c)

Moreover, all functions in this class have no comments because their names are understandable intuitive
It would look strange if I add a comment to only one function in this class